### ISSUE TITLE
Learning rate reported as 1e-12 in wandb, even when set to 1e-4

### ISSUE THREAD

**Issue by henriklied** [SWivid/F5-TTS#198]
No matter whether I set the learning rate in finetune-cli.py or not, wandb reports the LR as 1e-12.

`python3 finetune-cli.py --exp_name F5TTS_Base --batch_size_per_gpu 30000 --batch_size_type frame --max_samples 64 --grad_accumulation_steps 1 --max_grad_norm 1 --epochs 650 --num_warmup_updates 1 --save_per_updates 6255 --last_per_steps 10000 --dataset_name data_prepped_2`

I've added print statements both in `finetune-cli.py` and `trainer.py`, and both of these output `0.0001` when I don't specify a LR in the `finetune-cli.py`:
```
root@rough-life-passes-ice-01:~/dev/F5-TTS# python3 finetune-cli.py --exp_name F5TTS_Base --batch_size_per_gpu 30000 --batch_size_type frame --max_samples 64 --grad_accumulation_steps 1 --max_grad_norm 1 --epochs 650 --num_warmup_updates 1 --save_per_updates 6255 --last_per_steps 10000 --dataset_name data_prepped_2
**Learning rate is 0.0001**
wandb: Using wandb-core as the SDK backend. Please refer to https://wandb.me/wandb-core for more information.
wandb: Currently logged in as: henriklied. Use `wandb login --relogin` to force relogin
wandb: Tracking run with wandb version 0.18.5
wandb: Run data is saved locally in /root/dev/F5-TTS/wandb/run-20241021_073714-09mu3rfj
wandb: Run `wandb offline` to turn off syncing.
wandb: Syncing run F5TTS_Base
wandb: ⭐️ View project at
wandb: 🚀 View run at
**LR at trainer.py: 0.0001**
```

But still, on the wandb run, the LR is constantly reported as 1e-12:

![image](https://github.com/user-attachments/assets/51bd0086-6166-40ca-a56e-50f69079c470)

Any thoughts on what is going on here?

**Comment by SWivid** [comment]
not sure. maybe try  `--num_warmup_updates 1000`

**Comment by henriklied** [comment]
Tried several values for the `--num_warmup_updates` as well, but as of now, when resuming training at step 140k, it doesn't seem to have any effect on the learning rate reporting in wandb.

Are you able to reproduce this issue?

**Comment by manashpratim** [comment]
I am getting the same issue

**Comment by SWivid** [comment]
@henriklied  ah, i know. what is the amount of data you're using?
many it's just a normal case as we use a linear decay learning rate, and your training process just got finished?
the end factor of lr_scheduler is 1e-8, while 1e-12=1e-4*1e-8

if you want to have a long train, just set a large enough epoch e.g. 10000

**Comment by henriklied** [comment]
@SWivid That sounds probable! I'm currently running a very small scale 40 hour single speaker Norwegian dataset, and I'm at epoch 311/step 140k. 

Changing the --epochs to 10000 didn't do anything in terms of LR changes when resuming from my 140k step checkpoint, though.

**Comment by SWivid** [comment]
> @SWivid That sounds probable! I'm currently running a very small scale 40 hour single speaker Norwegian dataset, and I'm at epoch 311/step 140k.
> 
> Changing the --epochs to 10000 didn't do anything in terms of LR changes when resuming from my 140k step checkpoint, though.

yes, cuz you have a optimizer state in your finetuen ckpts.
you might comment out the `if` and use the part in `else` to let it go.
https://github.com/SWivid/F5-TTS/blob/795cb19e4fb37fe2c9ade85dc6a80f6d6286b775/model/trainer.py#L159-L172
or prune the ckpt file directly, removing the `step` key e.g.

**Comment by henriklied** [comment]
Thanks, @SWivid, that did the trick!
